packages/p2p/src/components/buy-sell/search-box.jsx

```diff
@@ -0,0 +1,89 @@
+import React from 'react';
+import { Field as FormField, Formik, Form } from 'formik';
+import { action } from 'mobx';
+import { Input, Icon } from '@deriv/components';
+import { isDesktop } from '@deriv/shared';
+import { localize } from 'Components/i18next';
+import { useStores } from 'Stores';
+import 'Components/buy-sell/search-box.scss';
+
+const SearchBox = () => {
+    const { buy_sell_store } = useStores();
+
+    const onSearch = ({ search }) => {
+        const search_result = [];
+
+        buy_sell_store.setSearchTerm(search.trim());
+
+        if (!search.trim()) {
+            buy_sell_store.setSearchResult([]);
+            return;
+        }
+
+        buy_sell_store.items.forEach(item => {
+            if (item.advertiser_details.name.toLowerCase().includes(search.toLowerCase().trim())) {
+                search_result.push(item);
+            }
+        });
+
+        if (search_result.length) {
+            buy_sell_store.setSearchResult(search_result);
+        } else {
+            buy_sell_store.setSearchResult([]);
+        }
+    };
+
+    const onSearchClear = setFieldValue => {
+        setFieldValue('search', '');
+        buy_sell_store.setSearchTerm('');
+        buy_sell_store.setSearchResult([]);
+    };
+
+    const onSearchKeyUp = submitForm => {
+        clearTimeout(typing_timer);
+
+        const typing_timer = setTimeout(
+            action(() => {
+                submitForm();
+            }),
```

Comment:
Any reason for this to be wrapped in `action`? Normally we'd use `action`s to tell MobX we're updating an observable inside the passed callback, in this case we're calling another function inside the passed callback.

Reply:
This is a remnant of trying to update the observable here but it's not needed anymore, as you said. Thanks for your review Aaron!